projects/plugins/boost/app/modules/cache/cache.php

```diff
@@ -0,0 +1,168 @@
+<?php
+
+class Boost_Cache {
+	private $cache_key     = false;
+	private $path_key      = false;
+	protected $request_uri = false;
+
+	public function __construct() {
+		$this->request_uri = isset( $_SERVER['REQUEST_URI'] )
+			? $_SERVER['REQUEST_URI'] // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized, WordPress.Security.ValidatedSanitizedInput.MissingUnslash
+			: false;
```

Comment:
You're right that the get() and set() functions might get different request_uri variables. I think we can store the MD5 of the request_uri directly.  We're using the MD5 of the path now for the cache filename. After all, the first thing `calculate_path()` does is get the md5 of the `$request_uri`.

Later, if we want to show a listing of the cache pages, I think we will need to record the path. That path can be sanitized in the same way that WP Super Cache does, and that's possible before WordPress loads. That info can either go in the cache file in a JSON array, or a separate sidecar file alongside each cache file. It will be escaped before displaying too.

Reply:
After thinking about it more, we still need the request_uri - it's used to calculate the cache key and the path to the cache file, and it can be sanitized. I've copied the sanitization code from WP Super Cache.